modules/tinymce/src/core/main/ts/api/html/DomParser.ts

```diff
@@ -80,154 +78,19 @@ interface DomParser {
   addNodeFilter: (name: string, callback: (nodes: AstNode[], name: string, args: ParserArgs) => void) => void;
   getNodeFilters: () => ParserFilter[];
   filterNode: (node: AstNode) => AstNode;
-  parse: (html: string, args?: ParserArgs) => AstNode;
+  parse: (html: string, args?: ParserArgs) => Node;
 }
 
 const DomParser = (settings?: DomParserSettings, schema = Schema()): DomParser => {
   const nodeFilters: Record<string, ParserFilterCallback[]> = {};
   const attributeFilters: ParserFilter[] = [];
-  let matchedNodes: Record<string, AstNode[]> = {};
-  let matchedAttributes: Record<string, AstNode[]> = {};
+  const matchedNodes: Record<string, AstNode[]> = {};
+  const matchedAttributes: Record<string, AstNode[]> = {};
 
   settings = settings || {};
   settings.validate = 'validate' in settings ? settings.validate : true;
   settings.root_name = settings.root_name || 'body';
 
-  const fixInvalidChildren = (nodes: AstNode[]) => {
```

Comment:
All of this logic shouldn't have been deleted, as we're going to need to retain it (or similar logic) long term.

- [ ] Revert this and either make it work or comment it out

Reply:
It's back, albeit unused right now, with comments saying to fold this logic into the new system.